tests/transforms/test_array_utils.py

```diff
@@ -132,3 +132,105 @@ def test_crop_array_raises_error(arr):
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((1, 6, 1, 6), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((2, 7, 2, 7), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [1, 1, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((0, 3, 0, 4), ((2, 0), (1, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (2, 2),
+                ((3, 5, 3, 5), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (3, 3),
+                ((2, 5, 2, 5), ((0, 0), (0, 0)))),
+            ])
+def test_content_extents(arr, shape, expected):
+    bounds = au.content_extents(arr, shape)
+    assert np.all(bounds == expected)
+
+
+@pytest.mark.parametrize(
+        ("arr", "shape"),
+        [
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 1, 0, 0],
+                    [0, 0, 1, 2, 2, 0, 0],
+                    [0, 0, 1, 1, 3, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5)),
+            ])
+def test_compare_crop_pad_and_extents(arr, shape):
+    """check that cropping and padding an image
+    is the same as getting the extents and
+    indexing the array (i.e. as would be applied to video frame)
+    """
+    cropped_padded = au.center_pad_2d(au.crop_2d_array(arr), shape)
+    extents, pad_width = au.content_extents(arr, shape)
+    indexed = np.pad(
+            arr[extents[0]:extents[1], extents[2]:extents[3]],
+            pad_width)
+    assert np.all(cropped_padded == indexed)
```

Comment:
I think np.testing.equal might be a better assertion function, either works

Reply:
That's not the name of a function. I assume you mean [`numpy.testing.assert_equal()`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_equal.html?highlight=assert_equal#numpy-testing-assert-equal)
If you have some experience or reason to share why you "think it might be a better assertion function" please share.

Reply:
Yes thats the function name, my bad.
Assert_equal searches for the first instance of inequality and raises an exception then. It also prints the inequality of the arrays which to me produces much more meaningful stack traces than getting a true or false from np.all.